src/main/java/seedu/address/storage/ImportManager.java

```diff
@@ -159,20 +166,37 @@ private String convertCsvContentsToJsonContents() throws IOException {
      * @param line csv line to be converted
      * @return String representation of a person in correct json format.
      */
-    private String convertLineToJsonPerson(String line) {
+    public String convertLineToJsonPerson(String line) throws IOException {
         String[] data = line.split(",");
         StringJoiner tagsJoiner = new StringJoiner("\", \"", "[\"", "\"]");
+        boolean areTagsValid = true;
         if (data.length > 6) {
             for (String tag : data[6].split(";")) {
+                areTagsValid = areTagsValid && isValidTagName((capitalizeWords(tag.trim())));
                 tagsJoiner.add(capitalizeWords(tag.trim()));
             }
         }
-        return String.format(
-                "  { \"studentId\": \"%s\", \"name\": \"%s\", \"parentPhoneNumberOne\": \"%s\", "
-                        + "\"parentPhoneNumberTwo\": \"%s\", \"email\": \"%s\", \"address\": \"%s\", \"tags\": %s,"
-                        + " \"formClass\": \"%s\" }",
-                data[0], capitalizeWords(data[1]), data[2], data[3], data[4], capitalizeWords(data[5]), tagsJoiner,
-                data[7]);
+
+        boolean isDataValid = (
+                isValidStudentId(data[0])
+                        && isValidName(capitalizeWords(data[1]))
+                        && isValidPhone(data[2])
+                        && isValidPhone(data[3])
+                        && isValidEmail(data[4])
+                        && isValidAddress(data[4])
+                        && areTagsValid
+                        && isValidClassName(data[7]));
+
+        if (!isDataValid) {
+            throw new IOException("Invalid data fields found in list, unable to merge!");
```

Comment:
Instead of chaining &&s, might be good to set exception message to something like
`invalid (FIELD) field found in (PERSON)` so that the user knows what to change

Reply:
Possible but would require bigger changes to what messages are being printed onto the console by the Commands themselves. (As of now the commands don't print anything other than MESSAGE_FAILURE)